### ISSUE TITLE
In Multiplayer the "Get 50 kills with XY weapon(tried on kuda) with an optic and 5 attachments equipped" doesn't work

### ISSUE THREAD

**Issue by LilTurix** [momo5502/boiii#471]
### Bug Type

Client

### BOIII Version

v0.0.1 (Latest)

### Environment (Operating System)

Microsoft Windows 10 (64-Bit)

### Reproducibility Rate

Medium: The bug can be reproduced intermittently

### Steps to Reproduce

1. ,
2. ,
3. ,

### Expected Result

Working camo challenge

### Actual Result

I tried to grind the camos for the kuda and the only camo i can't unlock for the weapon is the Bliss camo. I killed many players with it but it not count at at all

### Severity/Priority

Minor

**Comment by Brentdevent** [comment]
I have seen two people unlocking dark matter already in the discord, are you sure you're not overlooking something?

**Comment by LilTurix** [comment]
> I have seen two people unlocking dark matter already in the discord, are you sure you're not overlooking something?

No, for me some weapons count the kills, for example i have tried with the kn-44 it’s working without any problem but for the Locus it doesn’t even leveling up for me, same for the camo challenge it doesn’t count anything

**Comment by C0dyJb** [comment]
> I have seen two people unlocking dark matter already in the discord, are you sure you're not overlooking something?

They're probably using unlock all to get the camo.

**Comment by January-0002** [comment]
> > I have seen two people unlocking dark matter already in the discord, are you sure you're not overlooking something?
> 
> They're probably using unlock all to get the camo.

Nah, we both posted screenshots of legitimately unlocking them. Never had the issue with 6 attachment guns. I'll make a new players folder with a new profile and try to replicate tonight.

**Comment by Brentdevent** [comment]
> > I have seen two people unlocking dark matter already in the discord, are you sure you're not overlooking something?
> 
> No, for me some weapons count the kills, for example i have tried with the kn-44 it’s working without any problem but for the Locus it doesn’t even leveling up for me, same for the camo challenge it doesn’t count anything

Allright I have found the issue. You probably unlocked the weapons through the stats menu. Since it's not an actual unlock more of a workaround that tricks the game in thinking you unlocked them. It seems it does an additional check somewhere which causes the challenges/leveling not progressing. So for now you can disable `unlock all purchases` in the stats menu and unlock them legit with unlock tokens. Untill this gets fixed.
